### ISSUE TITLE
Flysystem resolver generates URL with double slash if cache_prefix is null

### ISSUE THREAD

**Issue by vroad** [liip/LiipImagineBundle#1121]
Flysystem resolver does not generate URL correctly if cache_prefix is not specified, because it does not remove leading from relative URL if no cache prefix is specified. If this is just not supported, why not throw errors?

I tried to put image files and filtered images in 2 different filesystems. They are actually subdirectory of a same S3 Bucket, so I could configure flysytem to use 1 filesystem instead, though.

https://github.com/liip/LiipImagineBundle/blob/4c321e023e06e5541d0312eeea5fbc33e758fea8/Imagine/Cache/Resolver/FlysystemResolver.php#L165-L174

### Preconditions

1. liip/imagine-bundle 2.1.0
2. Symfony 4.1.2

### Steps to reproduce

1. Create config files like this:

.env: 
```
# ...
UPLOADS_BUCKET=uploads
LIIP_IMAGINE_RESOLVER_ROOT_URL=http://localhost:9000/uploads/cache
```
oneup_flysystem.yaml:
```yaml
oneup_flysystem:
    adapters:
        uploads_adapter:
            awss3v3:
                client: aws.s3
                bucket: "%env(UPLOADS_BUCKET)%"
                prefix: uploads
        cache_adapter:
            awss3v3:
                client: aws.s3
                bucket: "%env(UPLOADS_BUCKET)%"
                prefix: cache
    filesystems:
        uploads:
            adapter: uploads_adapter
            alias: League\Flysystem\Filesystem
        cache:
            adapter: cache_adapter
```
config/packages/liip_imagine.yaml
```yaml
liip_imagine:
    enqueue: true
    loaders:
        uploads:
            flysystem:
                filesystem_service: oneup_flysystem.uploads_filesystem
    resolvers:
        default:
            flysystem:
                filesystem_service: oneup_flysystem.cache_filesystem
                root_url: "%env(LIIP_IMAGINE_RESOLVER_ROOT_URL)%"
                visibility: public
    data_loader: uploads
    driver: gd
    filter_sets:
        thumbnail:
            jpeg_quality: 100
            png_compression_level: 9
            filters:
                strip: ~
                thumbnail:
                    size: [200, 200]
                    mode: outbound
```
2. Call `FilterService::getUrlOfFilteredImage`
```php
$filePath = 'job/c50ae226-b6eb-4843-84df-d690af1e165d/562d264c-0dc1-4db6-b965-69834f35404e.jpeg'
$this->filterService->getUrlOfFilteredImage($filePath, 'thumbnail')
```

### Expected result

1. The method should return url without double slashes: `http://localhost:9000/uploads/cache/thumbnail/job/c50ae226-b6eb-4843-84df-d690af1e165d/562d264c-0dc1-4db6-b965-69834f35404e.jpeg`

### Actual result

1. The returned url has double slash: `http://localhost:9000/uploads/cache//thumbnail/job/c50ae226-b6eb-4843-84df-d690af1e165d/562d264c-0dc1-4db6-b965-69834f35404e.jpeg`

**Comment by gnutix** [comment]
I have the same issue on my setup. I've fixed it by either doing : 

```php
    public function resolve($path, $filter)
    {
        return sprintf(
            '%s/%s',
-            $this->webRoot,
-            $this->getFileUrl($path, $filter)
+            rtrim($this->webRoot, '/'),
+            ltrim($this->getFileUrl($path, $filter), '/')
        );
    }
```

or : 
```diff
    protected function getFileUrl($path, $filter)
    {
        // crude way of sanitizing URL scheme ("protocol") part
        $path = str_replace('://', '---', $path);

-        return $this->cachePrefix.'/'.$filter.'/'.ltrim($path, '/');
+        return ltrim($this->cachePrefix.'/'.$filter.'/'.$path, '/');
    }
```

Not sure which one would be the cleanest fix. I've not tried extensively, as "it works" with double slashes (though it's not "nice").

**Comment by dbu** [comment]
fixed in #1195
